### PR TITLE
Broker on mac support

### PR DESCRIPTION
**Pull request by xiangyan99**

### CODE REVIEW

File: sdk/identity/azure-identity-broker/azure/identity/broker/_browser.py [hunk 1]
```diff
@@ -160,6 +192,43 @@ def _get_app(self, **kwargs: Any) -> msal.ClientApplication:
...
+
+        if kwargs.get("enable_cae"):
+            client_applications_map = self._cae_client_applications
+            capabilities = ["CP1"]
+            token_cache = self._cae_cache
+
+        if not token_cache:
+            token_cache = self._initialize_cache(is_cae=bool(kwargs.get("enable_cae")))
+
+        if tenant_id not in client_applications_map:
```
Comment (pvaneck) [inline sdk/identity/azure-identity-broker/azure/identity/broker/_browser.py:220]: We should remove this conditional, right? Presumably, we want to override the cached app that has the broker enabled.

File: sdk/identity/azure-identity-broker/azure/identity/broker/_browser.py [hunk 2]
```diff
@@ -160,6 +192,43 @@ def _get_app(self, **kwargs: Any) -> msal.ClientApplication:
...
+            client_applications_map[tenant_id] = app_class(
+                client_id=self._client_id,
+                client_credential=self._client_credential,
+                client_capabilities=capabilities,
+                authority="{}/{}".format(self._authority, tenant_id),
+                azure_region=self._regional_authority,
+                token_cache=token_cache,
+                http_client=self._client,
+                instance_discovery=self._instance_discovery,
+                enable_broker_on_windows=True,
```
Comment (pvaneck) [inline sdk/identity/azure-identity-broker/azure/identity/broker/_browser.py:230]: Nit: might as well set windows to False in order to align with the method name?

File: sdk/identity/azure-identity-broker/CHANGELOG.md [hunk 3]
```diff
@@ -1,9 +1,11 @@
 # Release History
 
-## 1.2.1 (Unreleased)
+## 1.3.0b1 (Unreleased)
 
 ### Features Added
 
+- Added broker on MacOs support.
```
Comment (pvaneck) [inline sdk/identity/azure-identity-broker/CHANGELOG.md:7]:
```suggestion
- Added broker on macOS support.
```

**Review by pvaneck — COMMENTED** [review]
Let's also update the docstring to note that macOS is supported for the class. The README introduction should also be updated, as it also currently states that the credential is Windows only. 

How about going from:

```
Currently, only the Windows authentication broker, Web Account Manager (WAM), is supported. Users on macOS and Linux will be authenticated through a browser.
```
to 
```
Currently, only the following brokers are supported:

- Web Account Manager (WAM) on Windows
- Company Portal on macOS

Users on Linux will be authenticated through the browser.
```

**Comment by xiangyan99** [comment]
> Let's also update the docstring to note that macOS is supported for the class. The README introduction should also be updated, as it also currently states that the credential is Windows only.
> 
> How about going from:
> 
> ```
> Currently, only the Windows authentication broker, Web Account Manager (WAM), is supported. Users on macOS and Linux will be authenticated through a browser.
> ```
> 
> to
> 
> ```
> Currently, only the following brokers are supported:
> 
> - Web Account Manager (WAM) on Windows
> - Company Portal on macOS
> 
> Users on Linux will be authenticated through the browser.
> ```

Good catch!

File: sdk/identity/azure-identity-broker/azure/identity/broker/_browser.py [hunk 4]
```diff
@@ -32,8 +33,10 @@ class PopTokenRequestOptions(TokenRequestOptions):
 class InteractiveBrowserBrokerCredential(_InteractiveBrowserCredential):
     """Uses an authentication broker to interactively sign in a user.
 
-    Currently, only the Windows authentication broker, Web Account Manager (WAM), is supported. Users on macOS and Linux
-    will be authenticated through a browser.
+    Currently, only the following brokers are supported:
+    - Web Account Manager (WAM) on Windows
+    - Company Portal on macOS
```
Comment (scottaddie) [inline sdk/identity/azure-identity-broker/azure/identity/broker/_browser.py:38]: Does this mean the macOS broker can only be used by 1P customers?

**Review by pvaneck — APPROVED** [review]
Don't forget to also add the doc update to the README. 

Otherwise, lgtm! I manually tested on both Windows and Mac, and brokered auth was successful.

File: sdk/identity/azure-identity-broker/azure/identity/broker/_browser.py (hunk 4 above)
Comment (xiangyan99) [reply sdk/identity/azure-identity-broker/azure/identity/broker/_browser.py:38]: Company portal + Intune is Microsoft endpoint management solution that enables organizations to manage and secure their devices, applications, and data from a single platform.

It is not only used by Microsoft.

So I believe the answer is no. :)

**Pull request merged**